CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java

```diff
@@ -0,0 +1,494 @@
+
+            assertEquals(queryResults.get(GraphRecordStoreUtilities.DESTINATION + AnalyticConcept.VertexAttribute.RAW), "name2@domain2.com");
+            assertEquals(queryResults.get(GraphRecordStoreUtilities.DESTINATION + AnalyticConcept.VertexAttribute.TYPE), "Email");
+            assertEquals(queryResults.get(GraphRecordStoreUtilities.DESTINATION + AnalyticConcept.VertexAttribute.COMMENT), "TESTPARAMETERSPLUGIN");
+            assertEquals(queryResults.get(GraphRecordStoreUtilities.DESTINATION + TemporalConcept.VertexAttribute.LAST_SEEN), DateTimeFormatter.ISO_INSTANT.format(dtrStartEnd[1]).replace("Z", ".000Z"));
+
+        } catch (final PluginException ex) {
+            fail();
+            throw ex;
+        }
+    }
+
+    /**
+     * Test of query method, of class TestParametersPlugin. Tests not throwing
+     * runtime exception
+     */
+    @Test
+    public void testQueryCrash1() throws Exception {
+        System.out.println("test query crash1");
+
+        final TestParametersPlugin instance = new TestParametersPlugin();
+        final PluginParameters result = instance.createParameters();
+
+        final GraphRecordStore recordStore = new GraphRecordStore();
+        final DefaultPluginInteraction interaction = new DefaultPluginInteraction(null, null);
+
+        // Set plugin query name here before execution
+        result.getParameters().get(CoreGlobalParameters.QUERY_NAME_PARAMETER_ID).setStringValue("TESTPARAMETERSPLUGIN");
+
+        // Set plugin parameters here before execution
+        result.getParameters().get(TestParametersPlugin.CRASH_PARAMETER_ID).setBooleanValue(false);
+
+        try {
+            instance.query(recordStore, interaction, result);
+        } catch (final PluginException ex) {
+            fail();
+            throw ex;
+        }
+    }
+
+    /**
+     * Test of query method, of class TestParametersPlugin. Tests throwing
+     * runtime exception
+     */
+    @Test(expectedExceptions = RuntimeException.class)
+    public void testQueryCrash2() throws Exception {
+        System.out.println("test query crash2");
+
+        final TestParametersPlugin instance = new TestParametersPlugin();
+        final PluginParameters result = instance.createParameters();
+
+        final GraphRecordStore recordStore = new GraphRecordStore();
+        final DefaultPluginInteraction interaction = new DefaultPluginInteraction(null, null);
+
+        // Set plugin query name here before execution
+        result.getParameters().get(CoreGlobalParameters.QUERY_NAME_PARAMETER_ID).setStringValue("TESTPARAMETERSPLUGIN");
+
+        // Set plugin parameters here before execution
+        result.getParameters().get(TestParametersPlugin.CRASH_PARAMETER_ID).setBooleanValue(true);
+
+        try {
+            instance.query(recordStore, interaction, result);
+        } catch (final PluginException ex) {
```

Comment:
if the unit test throws Exception, you won't need a try-catch here (if all you're doing in the catch is throwing the exception anyway)